12/umbraco-cms/reference/content-delivery-api/extension-points/custom-property-editors-support.md

```diff
@@ -0,0 +1,227 @@
+---
+description: >-
+  Discover how to customize the Content Delivery API's response for your custom
+  property editors.
+---
+
+# Custom property editors support
+
+Out of the box, the Delivery API supports custom property editors, ensuring they are rendered alongside the built-in ones in Umbraco. However, if the output generated by your property editor isn't optimal for a headless context, you have the ability to customize the API response. This customization won't impact the Razor rendering, allowing you to tailor the Content Delivery API response according to your specific requirements.
+
+In this article, we'll look into how you can work with the IDeliveryApiPropertyValueConverter interface and implement custom [output expansion](../#output-expansion) for your custom property editors.
+
+## Prerequisite
+
+For the purpose of this example, it is assumed that you have created a custom property editor that stores the key of a content item. We will not dive into the details of creating a custom property editor and the necessary implementation steps within Umbraco for this article. If you need guidance on that, please refer to the [Creating a Property Editor](http://localhost:5000/s/OdQETpqkO0Kcv8KMquKL/tutorials/creating-a-property-editor) and [Property Value Converters](http://localhost:5000/s/OdQETpqkO0Kcv8KMquKL/extending/property-editors/property-value-converters) articles.
+
+## Implementation
+
+To implement the custom data values for your output in the Delivery API, we need to opt-in by implementing the `IDeliveryApiPropertyValueConverter` interface.
+
+The code example below showcases the implementation of a property editor called _"My.Custom.Picker"_. Our focus will be on the methods provided by the `IDeliveryApiPropertyValueConverter`, as they are responsible for customizing the Delivery API response. Towards the end of the file, you will find the response models that we will be using. The `IsConverter()` and `GetPropertyValueType()` methods are inherited from the `PropertyValueConverterBase` class, which is covered in the [Property Value Converters](http://localhost:5000/s/OdQETpqkO0Kcv8KMquKL/extending/property-editors/property-value-converters) article.
```

Comment:
There are some weird links to the https://docs.umbraco.com/umbraco-cms/tutorials/creating-a-property-editor https://docs.umbraco.com/umbraco-cms/extending/property-editors/property-value-converters/ articles. But I couldn't find them in the 12.latest space, so I had to link them through v11 folder which resulted in this. Some help from the Docs team is appreciated here!

P.S. I have no idea why it says that a sentence in ln.21 is longer than 24 words. I went though each of them with a word count